src/impl/keyence_high_speed_tcp.cpp

```diff
@@ -0,0 +1,197 @@
+#include "keyence/impl/keyence_high_speed_tcp.h"
+#include "boost/thread.hpp"
+#include "boost/atomic.hpp"
+
+#include "keyence/impl/messages/prepare_high_speed.h"
+#include "keyence/impl/messages/start_high_speed.h"
+#include "keyence/impl/messages/stop_high_speed.h"
+#include "keyence/impl/keyence_exception.h"
+#include "keyence/impl/ljv7_rawdata.h"
+
+#include "libsocket/epoll.hpp"
+
+#define KEYENCE_HIGHSPEED_NOTIFY_MESSAGE_SIZE 8
+
+namespace keyence
+{
+
+static void worker_main(boost::atomic_flag& flag,
```

Comment:
Since you are using c++11 is there a reason you haven't used the [std::atomic_flag](http://en.cppreference.com/w/cpp/atomic/atomic_flag)?